kerastuner/applications/resnet.py

```diff
@@ -12,14 +12,106 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-""" Basic network blocks that are used by Tunable Resnet.
+"""Hypertunable version of Resnet."""
 
-This code is based on https://github.com/keras-team/keras-applications.
-"""
-
-from tensorflow.keras import backend
+import tensorflow.keras as keras
 from tensorflow.keras import layers
-from tensorflow.keras import models
+from tensorflow.keras import optimizers
+from tensorflow.keras import backend
+
+from kerastuner.engine import hypermodel
+
+
+class HyperResnet(hypermodel.HyperModel):
+    """A ResNet HyperModel."""
+
+    def __init__(self, input_shape, num_classes, include_top=True):
+        self.input_shape = input_shape
+        self.num_classes = num_classes
+        self.include_top = include_top
+
+    def build(self, hp):
+        version = hp.Choice('version', ['v1', 'v2', 'next'], default='v2')
+
+        # Version-conditional hyperparameters.
+        with hp.name_scope(version):
+            conv3_depth = hp.Choice(
+                'conv3_depth',
+                [4] if version == 'next' else [4, 8],
+                default=4)
+            conv4_depth = hp.Choice(
+                'conv4_depth',
+                [6, 23] if version == 'next' else [6, 23, 36],
+                default=6)
+
+        # Version-conditional fixed parameters
+        preact = True if version == 'v2' else False
+        use_bias = False if version == 'next' else True
+
+
+        # Model definition.
+        bn_axis = 3 if backend.image_data_format() == 'channels_last' else 1
+        inputs = layers.Input(shape=self.input_shape)
+        x = inputs
+
+        # Initial conv2d block.
+        x = layers.ZeroPadding2D(padding=((3, 3), (3, 3)), name='conv1_pad')(x)
+        x = layers.Conv2D(
+            64,
+            7,
```

Comment:
Style nit: you can put all unnamed positional args on one line

Reply:
Done